src/main/java/sdu/sem2/se17/persistence/data/UserHandlerImpl.java

```diff
@@ -15,26 +21,135 @@ public UserHandlerImpl(DataSource dataSource) {
 
     @Override
     public Optional<User> create(User user) {
-        return Optional.empty();
+         Optional<User> result = Optional.empty();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(""" 
+                    INSERT INTO User (username, email, password, usertype)
+                    VALUES (?, ?, ?, ?)
+                    RETURNING *
+                """)
+        ) {
+            configureStatement(user, statement);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()){
+                    result = Optional.of(map(resultSet));
+                }
+            }
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+
+        return result;
     }
 
     @Override
     public Optional<User> read(long id) {
-        return Optional.empty();
+        Optional<User> result = Optional.empty();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(""" 
+                    SELECT * FROM \"User\"
+                    WHERE id = ?
+                    RETURNING *
+                """)
+        ) {
+            statement.setLong(1, id);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()){
+                    result = Optional.of(map(resultSet));
+                }
+            }
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+
+        return result;
     }
 
     @Override
     public void update(User user) {
-
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(""" 
+                    UPDATE User
+                    SET (name) = (username, password, email)
```

Comment:
Attributes must be referred to using valid SQL
```suggestion
                    SET (username, password, email) = 
```